src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -1797,15 +1789,12 @@ String fastFormat(double d) {
      * @param maxDigits the max digits
      */
     void setDigitList(Number number, boolean isNegative, int maxDigits) {
-
-        if (number instanceof Double) {
-            digitList.set(isNegative, (Double) number, maxDigits, true);
-        } else if (number instanceof BigDecimal) {
-            digitList.set(isNegative, (BigDecimal) number, maxDigits, true);
-        } else if (number instanceof Long) {
-            digitList.set(isNegative, (Long) number, maxDigits);
-        } else if (number instanceof BigInteger) {
-            digitList.set(isNegative, (BigInteger) number, maxDigits);
+        switch (number) {
+            case Double d -> digitList.set(isNegative, d, maxDigits, true);
+            case BigDecimal bd -> digitList.set(isNegative, bd, maxDigits, true);
+            case Long l -> digitList.set(isNegative, l, maxDigits);
+            case BigInteger bi -> digitList.set(isNegative, bi, maxDigits);
+            default -> {} // do nothing
```

Comment:
default case could also be thrown `AssertionError` if preferred.

Reply:
Or a simple `assert` statement, as throwing an Error could change the behavior

Reply:
Either of the 3 is fine with me. In this case, I think `AssertionError` is harmless, because AFAICT it is internally used by CompactNumberFormat only, and so throwing may prevent potential incorrect uses in the future.

Reply:
OK, then I think I am inclined to use `AssertionError`, assuming `DigitList` is defined to only accept those types.